src/components/Button.tsx

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import PropTypes from 'prop-types';
+
+/**
+ * Return our button and set its action
+ * @param className
+ * @param buttonText
+ * @param item
+ * @param action
+ * @constructor
+ */
+const Button = ({ className, buttonText, item, action }) => {
+  return (
+    <button
+      className={className}
+      onClick={() => action(item.id)}
+    >
+      {buttonText}
+    </button>
+  )
+}
+
+/**
+ * If nothing is passed in for these prop types
+ * the default value should be as below
+ */
+Button.defaultProps = {
+  buttonText: "Button",
+  className: "btn"
+}
+
+/**
+ * PropTypes is a npm package for setting the type of prop its
+ * expecting to receive for the component/ as of React version 15.5.0
+ * React.PropTypes has moved to package prop-types
+ */
+Button.propTypes = {
```

Comment:
Ahh, I see where PropTypes is used. This is interesting, have you used this on other projects? Given this is TypeScript you could probably do the same thing with an interface.

Reply:
@michaellouviere, are you familiar with this? Not sure what advantages it provides.